src/sasl/cyrsasl_scram.erl

```diff
@@ -37,17 +37,22 @@
 
 
 -record(state,
-        {step = 2              :: 2 | 4,
-         stored_key = <<"">>   :: binary(),
-         server_key = <<"">>   :: binary(),
-         username = <<"">>     :: binary(),
-         creds                 :: mongoose_credentials:t(),
-         auth_message = <<"">> :: binary(),
-         client_nonce = <<"">> :: binary(),
-         server_nonce = <<"">> :: binary(),
-         auth_module           :: ejabberd_gen_auth:t(),
-         sha                   :: sha()}).
-
+        {step = 2                  :: 2 | 4,
+         stored_key = <<"">>       :: binary(),
+         server_key = <<"">>       :: binary(),
+         username = <<"">>         :: binary(),
+         creds                     :: mongoose_credentials:t(),
+         auth_message = <<"">>     :: binary(),
+         client_nonce = <<"">>     :: binary(),
+         server_nonce = <<"">>     :: binary(),
+         auth_module               :: ejabberd_gen_auth:t(),
+         sha                       :: sha(),
+         plus_variant = none       :: plus_variant(),
+         plus_advertised           :: boolean(),
+         tls_last_message = <<"">> :: binary()
```

Comment:
Btw, `<<"">>` and `<<>>` are equivalent, and the second means less typing 😛

Reply:
True, less typing. Yet again went with consistency to how the empty default binaries were specified in the module. Do you think we should change all of them?

Reply:
Let's be consistent with the rest of the module and use `<<"">>` here.